components/admin_console/group_settings/group_details/group_teams_and_channels_row.jsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import GlobeIcon from 'components/svg/globe_icon';
+import LockIcon from 'components/svg/lock_icon';
+
+export default class GroupTeamsAndChannelsRow extends React.PureComponent {
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        type: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        implicit: PropTypes.bool,
+        hasChildren: PropTypes.bool,
+        collapsed: PropTypes.bool,
+        onRemoveItem: PropTypes.func.isRequired,
+        onToggleCollapse: PropTypes.func.isRequired,
+    }
+
+    removeItem = () => {
+        this.props.onRemoveItem(this.props.id, this.props.type);
+    }
+
+    toggleCollapse = () => {
+        this.props.onToggleCollapse(this.props.id);
+    }
+
+    render = () => {
+        let extraClasses = '';
+        let arrowIcon = null;
+        if (this.props.hasChildren) {
+            arrowIcon = (
+                <i
+                    className={'fa fa-caret-right' + (this.props.collapsed ? '' : ' open')}
+                    onClick={this.toggleCollapse}
+                />
+            );
+            extraClasses += ' has-clidren';
+        }
+
+        if (this.props.collapsed) {
+            extraClasses += ' collapsed';
+        }
+
+        let teamIcon = null;
+        let channelIcon = null;
+        switch (this.props.type) {
+        case 'public-team':
+            teamIcon = (
+                <div className='team-icon team-icon-public'>
+                    <i className={'fa fa-circle-o-notch'}/>
```

Comment:
Yes I'm changing it every where anyway, because is clearer for me to have literals without the `{}`.
@mkraft I think you are doing this , is something intended?